Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp

```diff
@@ -539,6 +574,204 @@ void GridTrackSizingAlgorithm::increaseSizesToAccommodateSpanningItems(const Gri
     increaseSizesToAccommodateSpanningItems<variant, TrackSizeComputationPhase::ResolveMaxContentMaximums>(gridItemsWithSpan, gridLayoutState);
 }
 
+template <TrackSizeComputationVariant variant>
+void GridTrackSizingAlgorithm::masonryIncreaseSizesToAccommodateSpanningItems(const StdMap<unsigned, MasonryMinMaxTrackSize> gridTrackSpans, WTF::StdMultimap<unsigned, GridItemWithSpan> definiteMultiTrackItems, GridLayoutState& gridLayoutState)
+{
+    auto increaseSizes = [&]<TrackSizeComputationPhase phase>(unsigned spanLength, WTF::StdMultimap<unsigned, MasonryMinMaxTrackSizeWithGridSpan> multiTrackItems)
+    {
+        Vector<GridTrack>& allTracks = tracks(m_direction);
+        for (const auto& trackIndex : m_contentSizedTracksIndex) {
+            GridTrack& track = allTracks[trackIndex];
+            track.setPlannedSize(trackSizeForTrackSizeComputationPhase(phase, track, TrackSizeRestriction::AllowInfinity));
+        }
+
+        Vector<WeakPtr<GridTrack>> growBeyondGrowthLimitsTracks;
+        Vector<WeakPtr<GridTrack>> filteredTracks;
```

Comment:
Can probably use WeakListHashSet instead

Reply:
This would require updating the grid implementation too, because the vectors are sent down to `distributeSpaceToTracks`.